### PR TITLE
Port profiler howtos from David Broman's blog to the documentation folder

### PR DESCRIPTION
**Pull request by davmason**

### CODE REVIEW

**Comment by noahfalk** [comment]
I would suggest putting all these files under path coreclr/Documentation/Profiling/davbr-blog-archive/... and then inserting a README in that directory. We can use the README to provide the right caveats that the blog might be out-of-date and readers will know what to expect.  Whether or not we want to update and correct the text or create new doc pages that reuse some of this text we can then do as a separate exercise at a time of our choosing.

**Comment by davmason** [comment]
Let me know what you think, I added dates to the entries and a readme blurb. FYI the one doc left in the top level Profiling folder is intentional, it was small enough I updated it to be applicable to .NET Core

File: Documentation/Profiling/davbr-blog-archive/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+David Broman's Profiling Blog Archive
+===========================
+
+David Broman was the original author of many profiler features in the .NET 4/4.5 timeframe. At the time he had many helpful blog posts, and even today are some of the most thorough documentation we have available on how to use the profiling APIs. They are starting to show their age a bit, and talk about desktop .NET since they were written before .NET Core existed. However they are still largely accurate and a valuable source of documentation for the profiling APIs.
```
Comment (noahfalk) [inline Documentation/Profiling/davbr-blog-archive/README.md]: The blog posts date back to 2005, which is around when desktop .NET 2.0 shipped.

File: Documentation/Profiling/Profiler Loading.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+There are two ways to load your profiler, with enviroment variables (cross-plat) or through the registry (Windows only)
+
+# Registry (Windows Only)
+Set the following environment variables:
+- `COR_ENABLE_PROFILING=1`
```
[the remark this answers is not in the document]
Comment (davmason) [reply Documentation/Profiling/Profiler Loading.md]: Thanks, don't know how I missed that

File: Documentation/Profiling/Profiler Loading.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+There are two ways to load your profiler, with enviroment variables (cross-plat) or through the registry (Windows only)
+
+# Registry (Windows Only)
+Set the following environment variables:
```
[the remark this answers is not in the document]
Comment (davmason) [reply Documentation/Profiling/Profiler Loading.md]: It was confusingly structured, those are the environment variables you have to set regardless. I was trying to say if you set those environment variables, we will look for the profiler following COM rules in the registry. I think it is more clearly worded now

File: Documentation/Profiling/davbr-blog-archive/README.md (hunk 1 above)
Comment (davmason) [reply Documentation/Profiling/davbr-blog-archive/README.md]: I didn't realize how old some of the posts were. The mention of which .NET version was not really relevant anyways, so I've changed it to just say he was an author of profiler features on desktop .NET

**Comment by davmason** [comment]
> 
> 
> I assume everything in the davbr-blog-archive is as-is and didn't scrutinize it. For the rest I made a few comments.

This is mostly true, I haven't done any rewording of his posts. The changes I did were to make the links work and I deleted some items that were clearly not necessary. For example a few posts had announcements at the top mentioning .NET 4.5 betas, or mentioning bugs that exist in pre 4.5 desktop versions but have since been fixed. Those were removed.

**Comment by mattwarren** [comment]
Thanks for doing this! I just went to https://blogs.msdn.com/b/davbr (which re-directed to https://blogs.msdn.microsoft.com/davbr/) and got very worried when I got the message:

> Oops! That page can’t be found.
> This page has been disabled.

David Broman's blog posts are invaluable for anyone writing a .NET Profiler!

**Pull request merged**